glusterd2/commands/volumes/volume-status.go

```diff
@@ -12,6 +12,41 @@ import (
 	"github.com/gorilla/mux"
 )
 
+func volumeStatusAllHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	volumes, e := volume.GetVolumes()
+	if e != nil {
+		logger.WithError(e).Error("Failed to get volumes")
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrVolNotStarted)
+		return
+	}
+	resp := createVolumeStatusAllResp(volumes)
+	restutils.SendHTTPResponse(ctx, w, http.StatusOK, resp)
+}
+
+func createVolumeStatusAllResp(volumes []*volume.Volinfo) *api.VolumeStatusAllResp {
+	var response api.VolumeStatusAllResp
+
+	for _, volinfo := range volumes {
+		if volinfo.State != volume.VolStarted {
+			return &response
+		}
+
+		s, err := volume.UsageInfo(volinfo.Name)
+		if err != nil {
+			return &response
+
+		}
+		size := createSizeInfo(s)
+
+		resp := createVolumeStatusResp(volinfo, &size)
+		response = append(response, *resp)
```

Comment:
With hundreds of volumes configured in cluster, wouldn't the response structure become too heavy for an API response? I'm still trying to understand what's the requirement of volume status all to be implemented through API.